src/KafkaFlow.OpenTelemetry/OpenTelemetryConsumerEventsHandler.cs

```diff
@@ -84,7 +84,16 @@ private static IEnumerable<string> ExtractTraceContextIntoBasicProperties(IMessa
 
         private static void SetConsumerTags(IMessageContext context, Activity activity)
         {
-            var messageKey = Encoding.UTF8.GetString(context.Message.Key as byte[]);
+            string messageKey = string.Empty;
+
+            try
+            {
+                messageKey = Encoding.UTF8.GetString(context.Message.Key as byte[]);
+            }
+            catch
+            {
+                // Do nothing
```

Comment:
Why do you need to use this try /catch ?

Reply:
KafkaFlow allows a messageKey to be null. When that happens, the GetString() method throws an exception

Reply:
Isnt better to check if the key is null? and avoid throwing errors?